### PR TITLE
[TTI] Add alignment argument to TTI for compress/expand support

### PR DESCRIPTION
**Pull request by nikolaypanchenko**
Since `llvm.compressstore` and `llvm.expandload` do require memory access, it's essential for some target to check if alignment is good to be able to lower them to target-specific instructions

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 1]
```diff
@@ -1609,3 +1609,28 @@ bool RISCVTTIImpl::isLSRCostLess(const TargetTransformInfo::LSRCost &C1,
                   C2.NumIVMuls, C2.NumBaseAdds,
                   C2.ScaleCost, C2.ImmCost, C2.SetupCost);
 }
+
+bool RISCVTTIImpl::isLegalMaskedCompressStore(Type *DataTy, Align Alignment) {
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1613]: I think this belongs in the patch that adds the RISCVISelLowering support. It's not legal for RISC-V without that.

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 2]
```diff
@@ -1609,3 +1609,28 @@ bool RISCVTTIImpl::isLSRCostLess(const TargetTransformInfo::LSRCost &C1,
...
+  auto *VTy = dyn_cast<VectorType>(DataTy);
+  if (!VTy || VTy->isScalableTy() || !ST->hasVInstructions())
+    return false;
+
+  Type *ScalarTy = VTy->getScalarType();
+  if (ScalarTy->isFloatTy() || ScalarTy->isDoubleTy())
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1619]: This is incorrect for Zve32f which doesn't support double or Zve64x and Zve32x that don't support float. There should be a function that handles this correctly in the gather/scatter TTI hook

Comment (nikolaypanchenko) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp]: thanks. I was looking for this before, but didn't know where I can find it. Will move the change to #83457

**Review by preames — APPROVED** [review]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**